tracer/src/Datadog.Trace/ClrProfiler/IAutomaticTracer.cs

```diff
@@ -0,0 +1,16 @@
+// <copyright file="IAutomaticTracer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.ClrProfiler
+{
+    internal interface IAutomaticTracer : ICommonTracer
+    {
+        object GetDistributedTrace();
+
+        void SetDistributedTrace(object trace);
```

Comment:
As far as I can tell, these always get and set the context as an `IReadOnlyDictionary<string, string>`. Can we use that here instead of `object` and get rid of the `as IReadOnlyDictionary<string, string>`?

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2020/commits/e1b26a25a9095a2a389a61a409e792c1b84a20b4

Reply:
Initially the object type was used on purpose due the rewriting, it's easier to refer to the object type as `ELEMENT_TYPE_OBJECT` than extracting the type signature for the `IReadOnlyDictionary<T,V>`.

https://github.com/DataDog/dd-trace-dotnet/blob/kevin/distributed-trace-rewriting/tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp#L1295

Changing this can break the rewriting...

Reply:
I think we should revert this change, and add a nice comment to avoid changing the type in future refactors :D

Reply:
We initially had to return object because this method was rewritten by the profiler, but that's not the case anymore. Now only `DistributedTracer.GetDistributedTracer` is rewritten so nothing prevents us from making `IAutomaticTracer` more strongly typed.

I thought long and hard about the value of keeping the parameter as object nonetheless in case we want to change the type in the future, but doing so would break the retro-compatibility so we would have to introduce a `IAutomaticTracer2` interface anyway.

Reply:
Ok, so then let's add the comment to `DistributedTracer.GetDistributedTracer` and leave the `IReadOnlyDictionary` here